roc.go

```diff
@@ -0,0 +1,53 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import "sort"
+
+type ByX struct {
+	x []float64
+	c []bool
+}
+
+func (a ByX) Len() int { return len(a.x) }
+func (a ByX) Swap(i, j int) {
+	a.x[i], a.x[j] = a.x[j], a.x[i]
+	a.c[i], a.c[j] = a.c[j], a.c[i]
+}
+func (a ByX) Less(i, j int) bool { return a.x[i] < a.x[j] }
+
+
+// ROC returns paired FPR and TPR values which describe the 
+// ROC curve treating y as a binary classifier for pos.
+func ROC(y []float64, pos []bool) ([]float64, []float64) {
+
+	if len(y) != len(pos) {
+		panic("inputs differ in length")
+	}
+
+	sort.Sort(ByX{y, pos})
+
+	var np float64
+	tpr := make([]float64, len(y)+1)
+	fpr := make([]float64, len(y)+1)
+	for i, u := range pos {
+		if u {
+			np += 1.0
+			tpr[i+1] = tpr[i] + 1.0
+			fpr[i+1] = fpr[i]
+		} else {
+			tpr[i+1] = tpr[i]
+			fpr[i+1] = fpr[i] + 1.0
+		}
+	}
+	nn := float64(len(y)) - np
```

Comment:
`nn := 1/(float(len(y)) - np)` and at L48 `fpr[i] *= nn`
`np = 1/np` and at L47 `tpr[i] *= np`.